docs/features/shared-volumes-with-NFS.md

```diff
@@ -0,0 +1,207 @@
+
+• Dirs are mounted only when user refers a remote dir when creating or attaching a volume
+• Unique labels for remote dirs
+• Assigned by the user and identifies a remote dir uniquely on a host
+• Selects a remote dir to create volumes
+• Using the same syntax as today – docker volume create –driver=vsphere-cloud –name=cloud-vol@dir_label …..
+• Creates the volume under /mnt/vmdk/dir_label/cloud_vol
+• The unique label is needed only if multiple remote dirs have been specified in the configuration
+• Plugin defaults to the one specified remote dir otherwise
+• There is no explicitly identified default remote dir to use if user doesn’t specify one
+
+
+# Mounting Remote Directories
+
+## Option #1
+– Plugin is provided the share/FS details via configuration
+• IP address of the server, path on server to mount, user/password, protocol, etc.
+– Plugin mounts the FS on a path (under /mnt/vmdk) that it uses to manage all datastores - /mnt/vmdk
+
+## Option #2
+– User mounts the share/FS on the VM and provides the path to the plugin via the config file or dynamically at docker volume create time
+– Plugin manages the provided path to create the volumes
+
+## Option #1 is preferred
+
+### Pros
+* Admin actions are reduced - doesn’t need to manage the remote directories on each host in a swarm
+* Plugin mounts remote dirs. as needed and only if volumes are referenced on those
+
+### Cons
+* Plugin needs to keep remote-dir configuration (in-memory) and possibly manage persistence
+
+
+# Handling remote dir configuration
+
+* On startup, plugin reads config and creates an in-memory map of datastores listed in the config
+* See sample config for what data is specified to mount a network share/FS
+* Each remote dir is identified by a unique label
+* User can provide a new remote dir configuration (via volume create opts) at volume creation
+* Plugin adds new the new dir to its in-memory map
+* User can refer the new dir via its unique label in subsequent volume ops
+* Does plugin need to persist dynamic entries in the config?
+* User identifies volumes by specifying <volume-name>@<remote-dir>
+
+
+# Volumes
+
+* Provisioned as a folder on the named network FS
+* /mnt/vmdk/<remote—dir-label>/<vol-path>/<vol-name>/
+* Volume create mkdir’s the volume name folder if not already present
+
+## Volume meta-data
+* Plugin maintains no in-memory state for either volumes or remote dirs. (except the configuration)
+* Volume met-data listed below are generated when requested via a Get() from Docker
+* Location – remote-dir, path within remote-dir
+* Type - volume type (NFS, EFS)
+* Capacity used in FS blocks
+* Current usage count
+* Create date/time
+* Modified time
+* Access time
+* Access permissions
+
+
+# Volume options and Tenancy
+
+## Volume create options
+   - Access perms, volume group, remote-dir (see remote dir config)
+## Tenancy
```

Comment:
The user is the one responsible for setting up the network shares and handles the items in the assumptions section earlier. No, there is no relation to the vmgroups as that type of tenancy isn't doable for these networked FS (the plugin doesn't interface with ESX at all for networked FS).